### DISCUSSION TITLE
/frigate/clips and /review - subfolder per camera or date

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#14817] [General]
These folders are rapidly filling up; I have 30,000 files. EXT4 and many other common filesystems don't perform as well with large numbers of files, nor do some network protocols.

Suggest that a folder is created per camera at least, or per yyyy-mm-dd like recordings.

Happy to look at sending a PR for 0.16 if acceptable.

**Comment by NickM-27** [comment]
The problem is there is no path for snapshots so changing the path is not simple change because if there is a new scheme how do we know where old and new snapshots are. Likely will happen at some point.

**Reply by leccelecce** [reply]
I guess a one-off migration to create camera folder and move all existing files into the right folders.

**Reply by NickM-27** [reply]
that would be pretty intensive since the necessary information is in the DB, but yeah that is one solution that could work

**Reply by leccelecce** [reply]
Ah, I hadn't appreciated there's a bit of mixture of things going on here. Some code stores thumbnails as base64 in the DB, some code stores an image path in the DB, some code just writes a file then assumes it knows where it is later.

Events don't look to store an image path, so at least I could move them into per-camera folders on startup, and update the code that writes/reads them to use the new location.
https://github.com/blakeblackshear/frigate/blob/fc0fb158d570b09fb0a97ff25f22faccf39a4658/frigate/api/event.py#L720 as one example seems to just assume where the file will be.
